### PR TITLE
Petruniv stable

### PR DESCRIPTION
**Pull request by vasinkevych**

### CODE REVIEW

File: src/app/containers/teachers-container/teachers/teacher-card/teacher-card.component.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  private maxAge = this.teachServise.checkAgeDate();
+  private editTeacher: AbstractControl;
+  private subscriptAvatar: any;
+
+  constructor(private teachServise: TeachersService,
+              private validServ: ValidationService,
+              private formBuilder: FormBuilder,
+              private notify: NotificationService) {
+                this.editTeacher = this.formBuilder.group({
+                   firstname: ['', ],
```
Comment (vasinkevych) [inline src/app/containers/teachers-container/teachers/teacher-card/teacher-card.component.ts]: can we replace  `''` with default values? So we can create one method e.g. 
```typescript
createFormWitDefaultValues() {
    this.editTeacher = this.formBuilder.group({
                   firstname: [this.teacher.firstname, ],
                   ......
}
```

Comment (vasinkevych) [reply src/app/containers/teachers-container/teachers/teacher-card/teacher-card.component.ts]: and call it oninit

File: src/app/containers/teachers-container/teachers/teacher-card/teacher-card.component.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    this.avatarImg = this.teacher.avatar
+      ? this.teacher.avatar
+      : '../../../assets/images/no-user-image.png';
+  }
+
+  submitEdit(event: Event): void {
+    event.preventDefault();
+    const data = this.editTeacher.value;
+    data.avatar = this.fileToUpload ? this.fileToUpload : this.teacher.avatar;
+    data.dateOfBirth = new Date(data.dateOfBirth).toISOString().slice(0, 10);
```
Comment (vasinkevych) [inline src/app/containers/teachers-container/teachers/teacher-card/teacher-card.component.ts]: maybe worth to use date-fns here

File: src/app/containers/teachers-container/teachers/teachers.component.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      )
+    ])
+  ]
+})
+export class TeachersComponent implements OnInit {
+  private columnsToDisplay: string[] = ['firstname', 'lastname', 'dateOfBirth'];
+  private expandedElement: Teacher | null;
+  private teachersList: MatTableDataSource<Teacher>;
+
+  constructor(private teachers: TeachersService) {}
```
Comment (vasinkevych) [inline src/app/containers/teachers-container/teachers/teachers.component.ts]: look like unused variable `private teachers: TeachersService`

File: src/app/containers/teachers-container/teachers/teachers.component.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  @ViewChild(MatPaginator, { static: true }) paginator: MatPaginator;
+
+
+  fillTable() {
+      this.teachersList = new MatTableDataSource<Teacher>(this.teachersPristine);
+      this.teachersList.paginator = this.paginator;
+  }
+
+  // function for sorting, trim() remove spaces
+  private applyFilter(filterValue: string): void {
```
Comment (vasinkevych) [inline src/app/containers/teachers-container/teachers/teachers.component.ts]: where do we use it?

File: src/app/services/teachers.service.ts [hunk 5]
```diff
@@ -1,81 +1,109 @@
...
-    });
+    return this.http
+      .get(`${this.BASE_URI}teachers`)
+      .subscribe((response: HttpGetReponse) => {
+        this.store.dispatch(teacherAction({ teachersList: response.data }));
+      });
   }
 
-  editTeacher(teacherId, data) {
-    return this.http.put(`${this.BASE_URI}admin/teachers/${teacherId}`, data, {
```
Comment (vasinkevych) [inline src/app/services/teachers.service.ts]: worth to create const for strings like `admin/teachers/${teacherId}` and add somewhere method eg `normalizeUrl/getUrl` in which will return correct full URL. Might be done in a separate task

File: src/app/services/teachers.service.ts [hunk 6]
```diff
@@ -1,81 +1,109 @@
...
-        this.store.dispatch(editTeacher({editedTeacher: response.body.data }));
-      } else {
-        this.notify.notifyFailure('Упс...щось пішло не так');
-      }
-    });
+  editTeacher(teacherId: number, data: Teacher) {
+    return this.http
+      .put<HttpPostPutReponse>(
+        `${this.BASE_URI}admin/teachers/${teacherId}`,
+        data,
```
Comment (vasinkevych) [inline src/app/services/teachers.service.ts]: see comment above

File: src/app/services/teachers.service.ts [hunk 7]
```diff
@@ -1,81 +1,109 @@
...
-        // @ts-ignore
-        this.store.dispatch(addOneTeacher({ teacher: response.body.data }));
-      } else {
-        this.notify.notifyFailure('Упс...щось пішло не так');
-      }
-    });
+  addTeacher(data: Teacher) {
+    return this.http
+      .post<HttpPostPutReponse>(`${this.BASE_URI}teachers`,
+      data,
```
Comment (vasinkevych) [inline src/app/services/teachers.service.ts]: see comment above

File: src/app/containers/teachers-container/teachers/teachers.component.ts (hunk 4 above)
Comment (SiJiP) [reply src/app/containers/teachers-container/teachers/teachers.component.ts]: <mat-form-field>
    <input matInput (keyup)="applyFilter($event.target.value)" placeholder="Пошук" class="search-input">
</mat-form-field>

File: src/app/containers/teachers-container/teachers/teacher-card/teacher-card.component.ts (hunk 1 above)
Comment (SiJiP) [reply src/app/containers/teachers-container/teachers/teacher-card/teacher-card.component.ts]: maybe now it will be work
earlier it was impossible

**Pull request closed without merging**